tests/functionaltests/Tests/CortanaTests.mm

```diff
@@ -54,36 +57,150 @@
+           public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>,
+                               IProtocolActivatedEventArgs,
+                               IActivatedEventArgs,
+                               IProtocolActivatedEventArgsWithCallerPackageFamilyNameAndData,
+                               IApplicationViewActivatedEventArgs,
+                               IViewSwitcherProvider> {
 
-                // Claim to be the implementation for the real system RuntimeClass for ProtocolActivatedEventArgs.
-                InspectableClass(RuntimeClass_Windows_ApplicationModel_Activation_ProtocolActivatedEventArgs,
-                                BaseTrust);
+               // Claim to be the implementation for the real system RuntimeClass for ProtocolActivatedEventArgs.
+               InspectableClass(RuntimeClass_Windows_ApplicationModel_Activation_ProtocolActivatedEventArgs, BaseTrust);
 
            public:
                MOCK_STDCALL_METHOD_1(get_Uri);
                MOCK_STDCALL_METHOD_1(get_Kind);
                MOCK_STDCALL_METHOD_1(get_PreviousExecutionState);
                MOCK_STDCALL_METHOD_1(get_SplashScreen);
+               MOCK_STDCALL_METHOD_1(get_CallerPackageFamilyName);
+               MOCK_STDCALL_METHOD_1(get_Data);
+               MOCK_STDCALL_METHOD_1(get_CurrentlyShownApplicationViewId);
+               MOCK_STDCALL_METHOD_1(get_ViewSwitcher);
            });
 
-// TODO: 8008558 Test handling foreground activation parameters from Cortana
+// Delegate for testing Cortana foreground activation
+// Use text value to guarantee it is the same argument we create
+@interface CortanaVoiceCommandForegroundTestDelegate : NSObject <UIApplicationDelegate>
+@property (nonatomic, readonly) NSMutableDictionary* methodsCalled;
+@end
+
+@implementation CortanaVoiceCommandForegroundTestDelegate
+- (id)init {
+    self = [super init];
+    if (self) {
+        _methodsCalled = [NSMutableDictionary new];
+    }
+    return self;
+}
+
+- (BOOL)application:(UIApplication*)application willFinishLaunchingWithOptions:(NSDictionary*)launchOptions {
+    ASSERT_TRUE(launchOptions[UIApplicationLaunchOptionsVoiceCommandKey]);
+    WMSSpeechRecognitionResult* result = launchOptions[UIApplicationLaunchOptionsVoiceCommandKey];
+    ASSERT_STREQ("CORTANA_TEST", [result.text UTF8String]);
+    [[self methodsCalled] setObject:@(YES) forKey:NSStringFromSelector(_cmd)];
+    return true;
+}
+
+- (BOOL)application:(UIApplication*)application didFinishLaunchingWithOptions:(NSDictionary*)launchOptions {
+    ASSERT_TRUE(launchOptions[UIApplicationLaunchOptionsVoiceCommandKey]);
+    WMSSpeechRecognitionResult* result = launchOptions[UIApplicationLaunchOptionsVoiceCommandKey];
+    ASSERT_STREQ("CORTANA_TEST", [result.text UTF8String]);
+    [[self methodsCalled] setObject:@(YES) forKey:NSStringFromSelector(_cmd)];
+    return true;
```

Comment:
Since you're in an instance method, you don't need to use the property accessor.
```_methodsCalled[NSStringFromSelector(_cmd)] = @(YES);```